### DISCUSSION TITLE
Autogluon on tabular classification problem

### DISCUSSION THREAD

**Discussion by mohdfaaf** [autogluon/autogluon#1549] [Q&A] [answered]
All 

I am quite new with Autogluon and python. I had attended some classes on python basic previously but it seems i havent really got myself to level that i wished to be at..  Still have to search for examples and copy paste and try run with hope no error. And usually there will be error :(. Thus i am stuck,... 

I have a  supervised classification problem framework that i have in mind - and would like to have a notebook that follows the framework. This framework will include the autoML part for model selection part for best one. I want to use Autogluon for the autoML. 

I am posting here as i have no idea how to get resource and help to do this ( as above )..  does anyone can help me for guidance..   i am also thinking to pay  ( not sure how much ) as option. 

Any input is welcome .. 

T

**Comment by Innixma** [comment, marked as answer]
You can [run AutoGluon locally](https://auto.gluon.ai/stable/index.html) or on most cloud services with hosted notebooks, such as AWS via SageMaker Notebooks. Free alternatives exist such as Google Colab [[AutoGluon Example Notebook on Colab](https://colab.research.google.com/drive/1F9idtZjiiyujLZ6YbqkOLArsbfDrWrUe?usp=sharing)].

**Reply by mohdfaaf** [reply]
Hi - thanks a lot for ur response. Your tips is very helpful. I managed to install the autogluon and run some codes successully. 

I believe i read somewhere mentioning that i can set the automl to explore based on model metric to what we set. 

I am working on  supervised learning for binary class which only has Pass or Fail. And typically every dataset that i worked with is imbalanced.. where Fail usually only 5% or less of the whole dataset ( 95% Pass 5% Fail ) . We dont use accuracy as our metric - instead we use precision % = Qty  true Fail predicted / Total predicted Fail and recall % = Qty true Fail predicted / Total true Fail. 

We want to find a model that can give highest recall% to predict Fail  given that precision% is kept to ~ 80%. 

Can we set the model exploration step ( automl ) to use this metric above? 

Thanks in advance. 

Faizal

**Reply by Innixma** [reply]
You can specify a custom metric to AutoGluon: https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-custom-metric.html

Or find a metric already defined in AutoGluon that roughly aligns with the one you are trying to optimize.
